L1Trigger/L1TGlobal/plugins/GtRecordDump.cc

```diff
@@ -396,6 +404,32 @@ namespace l1t {
           cout << "No Muon Data in this event " << endl;
         }
 
+        //Loop over Muon Showers
+        nObj = 0;
+        cout << " ------ Muons Showers --------" << endl;
+        if (muonShowers.isValid()) {
+          std::cout << "========= MuonShower BX index = " << i << "; min BX = " << m_minBx << "; max BX = " << m_maxBx
+                    << std::endl;
+          if (i >= muonShowers->getFirstBX() && i <= muonShowers->getLastBX()) {
+            for (std::vector<l1t::MuonShower>::const_iterator muShower = muonShowers->begin(i);
+                 muShower != muonShowers->end(i);
+                 ++muShower) {
+              cout << "  " << std::dec << std::setw(2) << std::setfill(' ') << nObj << std::setfill('0') << ")";
+              cout << "   MUS0 " << std::dec << std::setw(1) << muShower->isOneNominalInTime();
+              cout << ";  MUS1 " << std::dec << std::setw(1) << muShower->isOneTightInTime();
+              cout << ";  MUS2 " << std::dec << std::setw(1) << muShower->isTwoLooseDiffSectorsInTime();
+              cout << ";  MUSOOT0 " << std::dec << std::setw(1) << muShower->musOutOfTime0();
+              cout << ";  MUSOOT1 " << std::dec << std::setw(1) << muShower->musOutOfTime1();
+              cout << endl;
+              nObj++;
+            }
+          } else {
+            cout << "No MuonShowers stored for this bx " << i << endl;
+          }
+        } else {
+          cout << "No MuonShower Data in this event " << endl;
```

Comment:
Sorry, forgot to mention this, but here and everywhere else in the PR, you need to get rid of `std::cout`. Please use EDM logging facilities if this is needed: https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger

Reply:
Hi Andrew,
I'd be happy to remove the cout statements if necessary but first, Rick informed me that this file (GtRecordDump.cc) is not part of the official CMS workflow and all print statements in these files are just for those of us who work on the uGT emulator and because of this in the past it's been accepted that we can use the print statements rather than the EDM logger. If this is incorrect/no longer the case please let me know.
2nd, this file has a large number of cout statements that predate this pull request, so if cout should not be included then should I be going through and removing all of the preexisting instances of it as well?

Reply:
No, in this instance this is fine